NewForce-Cohort-9/tabloid-full-stack-full-stack-force
# Added User Profile Admin protection

# Description
Added admin protection to the client in 3 different places to ensure that there is always at least 1 active admin. Code can be modularized later if the protection will be needed in more places, but works for now.  The protection now exists in UserProfileList, UserProfileConfirm, and UserProfileEdit. 

[Trello Ticket #28](https://trello.com/c/GpoXNgs6)

# Testing Instructions
Add and commit any unsaved changes on your branch. Run git checkout main, git fetch --all, and `git checkout feat-user-profile-type`.
If you don't see the most recent version of my files, run `git pull origin feat-user-profile-type`

Inside of Visual Studio (purple), click the execute with debugger solid green button to launch the app.
In your terminal, navigate to where your package.json file is in you client folder. Run "npm start" in your terminal.

1. If not logged in, log in and use an admin account like [foo@bar.com](mailto:foo@bar.com) for the email parameter. Leave the password text box blank. Click "Login".
2. Click on User Profiles in the navbar. To test the functionality fully, you will need to deactivate a lot of your user profiles. You can do this by clicking the Deactivate button that is next to each active account in the list. At the confirmation page that you are directed to after clicking, click the red "Deactivate" button.
3. Repeat step 2 and deactivate every profile that has a UserType of "admin" until there is only 1 admin user left.
4. From the profile list, click the Deactivate button next to the last remaining admin. You should see a window alert appear with an error message indicating that the user must make someone else an admin before proceeding. 
5. Now, in case a user bypasses this protection from the user profile list, navigate to your last remaining admin's deactivate route manually by entering the URL in your web browser. For example, if your last user is the user with the ID of 6, navigate directly through your browser to `http://localhost:3000/profile/deactivate/6`. 
6. Once you are there, click the red Deactivate button. You should now see the same error message before that was available at the user profile list, which adds another layer of protection in case a user bypasses the protection from the user profile list.
7. Finally, go back to your user profile list, and this time click the "Edit" button next to your last remaining admin. When redirected to the `http://localhost:3000/profile/edit/{id}` route,  attempt to change the profile by selecting a User Type. Click the blue "Save" button. You should now see another error, that the user should make someone else an admin before changing this profile.
8. Ensure that you can reactivate and deactivate other user profiles (once you have more than one admin) as normal.

# Checklist:
[x] My code follows the style guidelines of this project
[x] I have performed a self-review of my code
[x] I have commented my code, particularly in hard-to-understand areas
[x] My changes generate no new warnings or errors
[x] I have added test instructions that prove my fix is effective or that my feature works
